### PR TITLE
Fix some code issues

### PR DESCRIPTION
**Pull request by cyyever**
Fixes #ISSUE_NUMBER

cc @jgong5 @mingfeima @XiaobingSuper @sanchitintel @ashokei @jingxu10

### CODE REVIEW

File: aten/src/ATen/cuda/CUDAEvent.h [hunk 1]
```diff
@@ -60,7 +60,9 @@ struct TORCH_CUDA_CPP_API CUDAEvent {
 
   CUDAEvent(CUDAEvent&& other) { moveHelper(std::move(other)); }
   CUDAEvent& operator=(CUDAEvent&& other) {
```
Comment (Skylion007) [inline aten/src/ATen/cuda/CUDAEvent.h:62]: Can both this and the move ctor be marked as noexcept for the perf boost?

File: torch/csrc/autograd/functions/tensor.cpp [hunk 2]
```diff
@@ -142,8 +142,7 @@ auto CopySlices::apply(variable_list&& inputs) -> variable_list {
       AT_ASSERT(res[i].defined());
       if (i == 0) {
         grad_slice.copy_(res[i]);
-        // NOLINTNEXTLINE(clang-analyzer-cplusplus.Move)
```
Comment (Skylion007) [inline torch/csrc/autograd/functions/tensor.cpp:145]: Pretty sure this was right before, result isn't used at all after this (this branch is never triggered again). I would revert the change for this file as this makes sense and this piece of code seems fairly perf critical.

File: torch/csrc/profiler/collection.cpp [hunk 3]
```diff
@@ -1129,12 +1129,16 @@ RecordQueue::getRecords(
     auto& queue = *subqueue_it.second;
     auto materialize = [&](auto& events) {
       for (auto& i : events) {
+        time_t start_time_ns;
+        if constexpr (std::is_same<
+                          typename std::remove_reference<decltype(i)>::type,
```
Comment (Skylion007) [inline torch/csrc/profiler/collection.cpp:1134]: If we are going to upgrade this, might as well use std::remove_reference_t

File: .clang-tidy [hunk 4]
```diff
@@ -37,6 +37,8 @@ modernize-*,
 performance-*,
 -performance-noexcept-move-constructor,
 -performance-unnecessary-value-param,
+misc-unused-using-decls,
```
Comment (Skylion007) [inline .clang-tidy:40]: Please properly order them to maintain alphabetical sort

**Review by Skylion007 — APPROVED** [review]

File: torch/csrc/cuda/Stream.cpp [hunk 5]
```diff
@@ -104,9 +103,7 @@ static PyObject* THCPStream_priority_range(
     PyObject* _unused,
     PyObject* noargs) {
   HANDLE_TH_ERRORS
-  // NOLINTNEXTLINE(cppcoreguidelines-init-variables)
-  int least_priority, greatest_priority;
-  std::tie(least_priority, greatest_priority) =
+  auto [least_priority, greatest_priority] =
```
Comment (albanD) [inline torch/csrc/cuda/Stream.cpp:106]: c++ 17 is amazing!

**Review by albanD — APPROVED** [review]
Thanks!

**Comment by albanD** [comment]
Can you rebase on latest master please? there seem to be some conflicts.
